File: validation schema/deliveryman.js
db.createCollection("deliverymans", {
  validator: {
    $jsonSchema: {
      required: ["name", "cpf", "plate", "store"],
      properties: {
        name: {
          bsonType: "string",
          maxLength: 64,
          description: "DeliveryMan Name.",
        },
        cpf: {
          bsonType: "string",
          maxLength: 11,
          description: "CPF.",
        },
        licensePlate: {
          bsonType: "string",
          maxLength: 7,
          description: "License plate.",
        },
        store: {
          bsonType: "object",
          description: "Store Id",
        },
      },
    },
  },
});

File: project-ifooddw/ms-deliveryman/src/main/java/com/ifooddw/deliveryman/model/DeliveryMan.java
package com.ifooddw.deliveryman.model;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import javax.persistence.Id;
import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document("deliverymans")
public class DeliveryMan {

    @Id
    private String id;
    private String name;
    private String cpf;
    private String licensePlate;
    @DBRef
    private Store store;

    @DBRef
    private List<Order> orders = new ArrayList<>();

}
